### DISCUSSION TITLE
error Command "v1.22.10" not found. When building Next.js app

### DISCUSSION THREAD

**Discussion by viknedus** [vercel/next.js#27449] [Help] [answered]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

16.4.0

### What browser are you using?

Safari

### What operating system are you using?

macOS Big Sur 11.4

### How are you deploying your application?

Vercel

### Describe the Bug

Using Yarn version 1.22.10

`yarn dev` runs the app pretty well during development.

#### Current behaviour

Unable to deploy a fresh Next.js app/repo to vercel

#### Expected behaviour

Deploy Next.js app/repo successfully 

### Vercel delopment environment

Everything is going fine until...

```log
16:06:46.444  	Cloning github.com/viknedus/next-app (Branch: main, Commit: b9cf787)
16:06:47.268  	Cloning completed: 823.678ms
16:06:47.297  	Analyzing source code...
16:06:48.293  	Installing build runtime...
16:06:50.544  	Build runtime installed: 2.250s
16:06:52.833  	Looking up build cache...
16:06:52.887  	Build Cache not found
16:06:53.970  	Installing dependencies...
16:06:54.318  	yarn install v1.22.10
16:06:54.376  	[1/5] Validating package.json...
16:06:54.378  	[2/5] Resolving packages...
16:06:54.602  	[3/5] Fetching packages...
16:07:13.705  	info fsevents@2.3.2: The platform "linux" is incompatible with this module.
16:07:13.706  	info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
16:07:13.739  	[4/5] Linking dependencies...
16:07:13.739  	warning " > better-react-spinkit@2.0.4" has unmet peer dependency "prop-types@^15.6.0".
16:07:13.740  	warning " > better-react-spinkit@2.0.4" has incorrect peer dependency "react@^16.2.0".
16:07:13.740  	warning " > better-react-spinkit@2.0.4" has unmet peer dependency "lodash@^4.17.4".
16:07:13.740  	warning " > styled-components@5.3.0" has unmet peer dependency "react-is@>= 16.8.0".
16:07:13.740  	warning "eslint-config-next > @typescript-eslint/parser > @typescript-eslint/typescript-estree > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
16:07:31.424  	[5/5] Building fresh packages...
16:07:31.853  	Done in 37.54s.
16:07:31.879  	Detected Next.js version: 11.0.1
16:07:32.063  	Running "`yarn run build`"
16:08:08.494  	yarn run v1.22.10
16:08:08.530  	error Command "v1.22.10" not found.
16:08:08.531  	info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
16:08:08.549  	Error: Command "`yarn run build`" exited with 1
```

### Expected Behavior

App Deployed Successfully on Vercel 

## Steps to Reproduce 

1. Create or import New Next.js repo on Github
2. Deploy Next.js app/repo as described [here](https://nx.dev/latest/react/guides/nextjs#deploying-to-vercel, "Deploying to vercel")
3. Deploy error

**Comment by advaiyalad** [comment]
@viknedus Could you please post your package.json? It seems like you modified the build script to be something else.

**Comment by viknedus** [comment]
@PythonCreator27 Sorry for late reply, here is my package.json. 
```
{
  "name": "next-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@material-ui/core": "^4.11.4",
    "@material-ui/icons": "^4.11.2",
    "better-react-spinkit": "^2.0.4",
    "email-validator": "^2.0.4",
    "firebase": "^8.6.8",
    "moment": "^2.29.1",
    "next": "11.0.1",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-firebase-hooks": "^3.0.4",
    "styled-components": "^5.3.0",
    "timeago-react": "^3.0.2"
  },
  "devDependencies": {
    "eslint": "7.29.0",
    "eslint-config-next": "11.0.1"
  }
}
```

**Comment by ijjk** [comment]
Hi, it doesn't look like the above `package.json` is using the the mentioned steps from https://nx.dev/latest/react/guides/nextjs#deploying-to-vercel please provide a minimal reproduction to allow investigating!

**Comment by viknedus** [comment]
Hi @ijjk, the `package.json` file was not tempered with. Everything in it was internally system generated. When i run `yarn dev` locally, the app builds successfully locally. If you look at the Vercel deploy error code, you will see that the problem might be generated from `yarn run build`

The 'build command' and 'output directory' was not modified nor overridden.

I simply pushed my code to github and then deployed to vercel with the errors stated above.

**Comment by ijjk** [comment]
@viknedus I say it looks like the provided guide wasn't followed since the dependencies used in the guide aren't provided in the `package.json` above. Can you provide a repo with a minimal reproduction of this error? We aren't able to reproduce this with the info provided.

**Comment by viknedus** [comment, marked as answer]
Hello @ijjk, Here is a repo to help reproduce the error: https://github.com/viknedus/WhatsApp-v2ish

**Reply by ijjk** [reply]
Just checked out the repo and it looks like you are importing a file that doesn't exist, here is the error message shown when building locally with `yarn build`:

```
yarn run v1.22.10
$ next build
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
info  - Checking validity of types  
info  - Creating an optimized production build  
Failed to compile.

ModuleNotFoundError: Module not found: Error: Can't resolve '../firebase.js' in '/tmp/WhatsApp-v2ish/pages'


> Build error occurred
Error: > Build failed because of webpack errors
    at /tmp/WhatsApp-v2ish/node_modules/next/dist/build/index.js:15:924
    at async Span.traceAsyncFn (/tmp/WhatsApp-v2ish/node_modules/next/dist/telemetry/trace/trace.js:6:584)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by viknedus** [reply]
@ijjk `firebase.js` does exist. It's committed in `.gitignore`. Should i remove it from `.gitignore` and send the updated link for you to try again?

**Reply by viknedus** [reply]
@ijjk updated: https://github.com/viknedus/WhatsApp-v2ish
Run it again. Thanks.

**Reply by ijjk** [reply]
@viknedus it looks like you customized your build command for your project at https://vercel.com/viknedus/whats-app-v2ish/settings if you uncheck the override toggle next to the build command setting and use the default command it should work alright

**Reply by viknedus** [reply]
That worked! The override toggle was the problem. Thanks @ijjk 👍🏾
